language_generators/scala-defref-extractor/src/main/scala/io/bazeltools/buildgen/scaladefref/ScalaSourceEntityExtractor.scala

```diff
@@ -746,9 +853,12 @@ object ScalaSourceEntityExtractor {
         log(s"Name($n)")
         if (n.nonEmpty) referTo(nm)
         else unitEnv
-      case Pkg.Object(mods, name, template) =>
+      case Pkg.Object(mods, name, template @ Template(_, init, _, stats)) =>
         log(s"package object: mods = $mods, name = $name, template = $template")
-        define(name) *> inside(name, inspect(template))
```

Comment:
I see `define(name)` is removed. I wonder if any `Object` needs to define that name, or can we always get away with just the members?

I guess we do, since you could have:

```scala
object EmptyIt extends Iterator[Nothing] {
...
}

callSomeIteratorFn(EmptyIt)
```
but that pattern is basically never used for package objects I guess... but I suppose it could be?

Anyway, we can add tests for the thing we need later if it causes problems.

Reply:
Yep we need a test around this and some more concrete argument noted in the test. And thats right we could have it with package objects but it feels rarer. Vs the other side where it causes huge links